src/components/map/sectors/MapSectorsList.vue

```diff
@@ -1,38 +1,138 @@
 </template>
 
 <script setup lang="ts">
 import VectorSource from 'ol/source/Vector';
 import type { ShallowRef } from 'vue';
-import type { Map } from 'ol';
+import type { Map, MapBrowserEvent } from 'ol';
 import { Fill, Stroke, Style } from 'ol/style';
-import MapSector from '~/components/map/sectors/MapSector.vue';
+import MapVatglassesPosition from '~/components/map/sectors/MapVatglassesPosition.vue';
 import VectorImageLayer from 'ol/layer/VectorImage';
+import { useStore } from '~/store';
+import MapSector from '~/components/map/sectors/MapSector.vue';
+
+import { initVatglasses } from '~/utils/data/vatglasses';
+import type { VatglassesSectorProperties } from '~/utils/data/vatglasses';
+
+import type { Pixel } from 'ol/pixel';
+import { useMapStore } from '~/store/map';
+
 
 let vectorLayer: VectorImageLayer<any>;
 const vectorSource = shallowRef<VectorSource | null>(null);
 provide('vector-source', vectorSource);
 const map = inject<ShallowRef<Map | null>>('map')!;
 const dataStore = useDataStore();
+const store = useStore();
+const mapStore = useMapStore();
+
 
 const firs = computed(() => {
     const list = dataStore.vatspy.value!.data.firs;
     const firs = list.map(fir => ({
         fir,
         atc: dataStore.vatsim.data.firs.value.filter(x => x.firs.some(x => x.boundaryId === fir.feature.id && (fir.icao === x.icao || (fir.callsign && fir.callsign === x.callsign)))) ?? [],
     }));
-
     return firs.filter((x, xIndex) => !firs.some((y, yIndex) => y.fir.icao === x.fir.icao && x.fir.feature.id === y.fir.feature.id && yIndex < xIndex));
 });
 
+
+const sectorsAtClick = shallowRef<VatglassesSectorProperties[]>([]);
+const getCoordinates = ref([0, 0]);
+const vatglassesPopupIsShown = ref(false);
+
+let lastEventPixel: Pixel | null = null;
+async function handleClick(e: MapBrowserEvent<any>) {
+    // TODO: don't show popup when clicked target has an aircraft
+    const eventPixel = map.value!.getPixelFromCoordinate(e.coordinate);
+
+    if (lastEventPixel && lastEventPixel[0] === eventPixel[0] && lastEventPixel[1] === eventPixel[1]) {
+        // same location, close popup
+        sectorsAtClick.value = [];
+        vatglassesPopupIsShown.value = false;
+        lastEventPixel = null;
+        return;
+    }
+    lastEventPixel = eventPixel;
+    const featureSectors = map.value!.getFeaturesAtPixel(eventPixel, {
+        hitTolerance: 0, // we use 6 instead of 5 because of the aircraft icons size, it is just for cosmetic reasons
+        layerFilter: layer => layer.getProperties().type === 'sectors',
+    });
+
+    const sectors: VatglassesSectorProperties[] = [];
+    featureSectors.map(feature => {
+        const properties = feature.getProperties() as VatglassesSectorProperties;
+        sectors.push(properties);
+    });
+
+    sectorsAtClick.value = sectors;
+
+    getCoordinates.value = e.coordinate;
+    vatglassesPopupIsShown.value = !!sectorsAtClick.value.length;
+}
+
+watch(() => mapStore.extent, () => {
```

Comment:
Could we use attachMoveEnd instead maybe? I'm afraid extent will be changing too often